File: backend/src/main/java/com/sistemacafeplanob/backend/entity/Usuario.java
package com.sistemacafeplanob.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "USUARIO")
@Data
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "PESSOA_ID", nullable = false, unique = true)
    private Pessoa pessoa;

    @Column(name = "USUARIO_LOGIN", unique = true, nullable = false)
    private String usuarioLogin;

    @Column(unique = true, nullable = false)
    private String email;

    private String senha;

    @CreationTimestamp
    private LocalDateTime data_criacao;
}

File: frontend/src/app/models/usuario.ts
export interface Pessoa {
  id?: number;
  nome: string;
  cpf: string;
  telefone?: string;
}

export interface Usuario {
  id?: number;
  pessoa?: Pessoa;
  usuarioLogin: string;
  email: string;
  senha?: string;
}

export interface LoginRequest {
  login: string;
  senha: string;
}
